Lib/Protocols/IdCompressorZLib.pas

```diff
@@ -172,7 +172,11 @@ procedure TIdCompressorZLib.InternalDecompressStream(
     repeat
       LZstream.next_out := @outBuffer[0];
       LZstream.avail_out := bufferSize;
-      DCheck(inflate(LZstream,Z_NO_FLUSH));
+      zresult := inflate(LZstream, Z_NO_FLUSH);
+      if zresult <> Z_BUF_ERROR then
+      begin
+        DCheck(zresult);
+      end;
       outSize := bufferSize - LZstream.avail_out;
       AOutStream.Write(outBuffer, outSize);

```

Comment:
If `Z_BUF_ERROR` were to be ignored, then presumably `outSize` will be 0 in that situation, so moving `AOutStream.Write()` inside the `if` block would make sense, yes?

Reply:
I think thers's no need to do

Reply:
I think thers's no need to do so